test/Test/Time/Units.hs

```diff
@@ -59,3 +60,21 @@ spec_Units = do
             floorUnit @Day (Time $ 5 :% 2) `shouldBe` 2
         it "returns 42ps when floor integer" $
             floorUnit (ps 42) `shouldBe` 42
+    describe "Formatting tests" $ do
+        it "4000 minutes should be formatted without ending-zeros" $
+            seriesF @'[DayUnit, HourUnit, MinuteUnit, SecondUnit] (minute 4000) `shouldBe` "2d18h40m"
+        it "4000 minutes should be formatted without beginning-zeros" $
+            seriesF @'[WeekUnit, DayUnit, HourUnit, MinuteUnit] (minute 4000) `shouldBe` "2d18h40m"
+        it "3601 sec should be formatted without middle-zeros" $
+            seriesF @'[HourUnit, MinuteUnit, SecondUnit] (sec 3601) `shouldBe` "1h1s"
+        it "works on rational nums" $
+            seriesF @'[HourUnit, SecondUnit, MillisecondUnit] (Time @MinuteUnit $ 3 :% 2) `shouldBe` "90s"
+        it "works without minutes formatting" $
+            seriesF @'[DayUnit, MinuteUnit, SecondUnit] (minute 4000) `shouldBe` "2d1120m"
+
+        it "4000 minutes should be formatted like 2d18h40m" $
+            unitsF (minute 4000) `shouldBe` "2d18h40m"
+        it "42 fortnights should be formatted like 42fn" $
+            unitsF (fortnight 42) `shouldBe` "42fn"
+        it "empty when receive zero time" $
+            unitsF (Time @HourUnit 0) `shouldBe` ""
```

Comment:
Personally I don't like this behavior. I didn't think of it. But I think that if `0` time is passed to `seriesF` formatter we want to use `0smth`. So the questions is what you would like to see in case you call:

```haskell
seriesF @'[Hour, Second] (minute 0)
```

1. `0h`
2. `0m`
3. `0s`

I think this deserves separate issue. I will create it.